tests/databases/commands_tests.py

```diff
@@ -508,3 +520,68 @@ def test_import_v1_rollback(self, mock_import_dataset):
         # verify that the database was not added
         new_num_databases = db.session.query(Database).count()
         assert new_num_databases == num_databases
+
+
+class TestTestConnectionDatabaseCommand(SupersetTestCase):
+    @mock.patch("superset.databases.dao.DatabaseDAO.build_db_for_connection_test")
+    @mock.patch("superset.databases.commands.test_connection.event_logger")
+    def test_connection_db_exception(
+        self, mock_event_logger, mock_build_db_for_connection_test
+    ):
+        """Test that users can't export databases they don't have access to"""
+        database = get_example_database()
+        mock_build_db_for_connection_test.side_effect = Exception(
+            "An error has occurred!"
+        )
+        db_uri = database.sqlalchemy_uri_decrypted
+        json_payload = {"sqlalchemy_uri": db_uri}
+        command_without_db_name = TestConnectionDatabaseCommand(
+            security_manager.find_user("admin"), json_payload
+        )
+
+        with self.assertRaises(DatabaseTestConnectionUnexpectedError):
+            command_without_db_name.run()
```

Comment:
Now that we're using `pytest`, it's better to run tests using plain `assert`s instead, since it gives more context when the test fails. Also, we can check that the exception raised has the right error message:

```suggestion
        with pytest.raises(DatabaseTestConnectionUnexpectedError) as excinfo:
            command_without_db_name.run()
         assert str(excinfo.value) == (
             "Unexpected error occurred, please check your logs for details"
         )
```

Same in the other 2 tests below.